### PR TITLE
Add new CSI translation APIs required by CSI sidecars and unit tests

### PR DESCRIPTION
**Pull request by ddebroy**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
Adds an extra pair of APIs to the CSI translation library so that migration scenarios initiated outside of in-tree (i.e. from CSI side car containers) like provisioning can use the library.

**Which issue(s) this PR fixes**:
Related to https://github.com/kubernetes/enhancements/issues/625

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

File: staging/src/k8s.io/csi-translation-lib/translate.go [hunk 1]
```diff
@@ -76,18 +76,27 @@ func TranslateCSIPVToInTree(pv *v1.PersistentVolume) (*v1.PersistentVolume, erro
...
 		}
 	}
 	return false
 }
 
-// GetCSINameFromIntreeName maps the name of a CSI driver to its in-tree version
-func GetCSINameFromIntreeName(pluginName string) (string, error) {
+// SupersedesInTreePlugin tests whether there exists an in-tree plugin with logic
+// to migrate to the CSI plugin with given csiPluginName
+func SupersedesInTreePlugin(csiPluginName string) bool {
```
Comment (davidz627) [inline staging/src/k8s.io/csi-translation-lib/translate.go]: We need to come up with better names (all of us in general, not just this). 

This is the CSI direction of `IsMigratableByName` and I think it should be evident through the naming.
Maybe they should be `IsIntreePluginMigratableByName` and `IsCSIDriverMigratableByName` instead?
WDYT?

Naming things is hard.

Comment (ddebroy) [reply staging/src/k8s.io/csi-translation-lib/translate.go]: API names are indeed hard! The one thing that bothers me about `IsCSIDriverMigratableByName` is: the CSI Driver is not migrat"able" to something - it's what the in-tree plugin has been migrat"ed" to. 

I was leaning towards `IsMigratedCSIDriverByName` and `IsMigratableIntreePluginByName` but I guess that does not align with the desired convention. If not, I will go ahead with `IsIntreePluginMigratableByName` and `IsCSIDriverMigratableByName`

Comment (davidz627) [reply staging/src/k8s.io/csi-translation-lib/translate.go]: I think your suggestion (IsMigratedCSIDriverByName and IsMigratableIntreePluginByName) makes sense. Thanks :+1:

**Review by saad-ali — APPROVED** [review]
/lgtm
/approve

**Pull request merged**